src/node/rpc/frontend.h

```diff
@@ -613,6 +431,194 @@ namespace ccf
       }
     }
 
+  public:
+    RpcFrontend(
+      Store& tables_,
+      HandlerRegistry& handlers_,
+      ClientSignatures* client_sigs_ = nullptr) :
+      tables(tables_),
+      nodes(tables.get<Nodes>(Tables::NODES)),
+      client_signatures(client_sigs_),
+      handlers(handlers_),
+      pbft_requests_map(
+        tables.get<pbft::RequestsMap>(pbft::Tables::PBFT_REQUESTS)),
+      consensus(nullptr),
+      history(nullptr)
+    {}
+
+    void set_sig_intervals(size_t sig_max_tx_, size_t sig_max_ms_) override
+    {
+      sig_max_tx = sig_max_tx_;
+      sig_max_ms = std::chrono::milliseconds(sig_max_ms_);
+      ms_to_sig = sig_max_ms;
+    }
+
+    void set_cmd_forwarder(
+      std::shared_ptr<enclave::AbstractForwarder> cmd_forwarder_) override
+    {
+      cmd_forwarder = cmd_forwarder_;
+    }
+
+    void open() override
+    {
+      std::lock_guard<SpinLock> mguard(lock);
+      is_open_ = true;
+
+      handlers.init_handlers(tables);
+    }
+
+    bool is_open() override
+    {
+      std::lock_guard<SpinLock> mguard(lock);
+      return is_open_;
+    }
+
+    /** Process a serialised command with the associated RPC context
+     *
+     * If an RPC that requires writing to the kv store is processed on a
+     * backup, the serialised RPC is forwarded to the current network primary.
+     *
+     * @param ctx Context for this RPC
+     * @returns nullopt if the result is pending (to be forwarded, or still
+     * to-be-executed by consensus), else the response (may contain error)
+     */
+    std::optional<std::vector<uint8_t>> process(
```

Comment:
Note: in the Raft case, `process()` is now pretty much a wrapper for `process_command()`. 
As we move more and more logic down to `process_command()` depending on which handler is neing executed, `process_command()` might be renamed to `process()` and the balance of the universe will be restored.